js/widgets/loader.js

```diff
@@ -119,50 +113,30 @@ return $.widget( "mobile.loader", {
 		message = msgText || ( loadSettings.text === false ? "" : loadSettings.text );
 
 		// prepare the dom
-		$html.addClass( "ui-loading" );
+		this._addClass( html, "ui-loading" );
 
 		textVisible = loadSettings.textVisible;
 
 		// add the proper css given the options (theme, text, etc)
 		// Force text visibility if the second argument was supplied, or
 		// if the text was explicitly set in the object args
-		this.element.attr( "class", loaderClass +
-			" ui-corner-all ui-body-" + theme +
-			" ui-loader-" + ( textVisible || msgText || theme.text ? "verbose" : "default" ) +
-			( loadSettings.textonly || textonly ? " ui-loader-textonly" : "" ) );
-
-		// TODO verify that jquery.fn.html is ok to use in both cases here
-		//      this might be overly defensive in preventing unknowing xss
-		// if the html attribute is defined on the loading settings, use that
-		// otherwise use the fallbacks from above
-		if ( loadSettings.html ) {
-			this.element.html( loadSettings.html );
-		} else {
-			this.element.find( "h1" ).text( message );
-		}
+		this._removeClass( "ui-loader-verbose ui-loader-default ui-loader-textonly" )
+			._addClass( "ui-loader-" +
+			( textVisible || msgText || theme.text ? "verbose" : "default" ) +
+			( loadSettings.textonly || textonly ? " ui-loader-textonly" : "" ),
+				"ui-body-" + theme );
```

Comment:
This function takes the swatch to be shown as an argument. The theme extension only handles ```_create()``` and ```_setOptions()```.